server/handler.go

```diff
@@ -92,12 +92,21 @@ func (h *Handler) NewConnection(c *mysql.Conn) {
 		h.sel.ClientConnected()
 	}
 
+	h.sm.InitialSession(c)
```

Comment:
Open to suggestions here...maybe these `processlist` interactions belong in `SessionManager` itself, and only `BeginQuery`/`EndQuery` on the `Context.ProcessList` belong in Handler? As is these are always paired up and we're violating law of demeter or something like it by mucking around with SessionManager's ProcessList.

Reply:
Same here :).